src/shared/treePaths/useTreePaths.js

```diff
@@ -34,37 +38,39 @@ export function useTreePaths(passedPath) {
     { suspense: false }
   )
 
-  const branch = urlBranch && decodeURIComponent(urlBranch)
-  const ref = urlRef && decodeURIComponent(urlRef)
-  const path = urlPath && decodeURIComponent(urlPath)
-  const filePaths = getFilePathParts(passedPath || path)
-  const defaultBranch = repoOverview?.defaultBranch
+  const treePaths = useMemo(() => {
```

Comment:
Same with this memo, should refrain from using `useMemo` unless it's absolutely required to create a stable reference OR there are massive performance implications, and for this function there's not a lot of complexity imo that would have a major performance impact.

Reply:
Also required to prevent infinite re-renders. I spent many hours dealing with this guy.

Reply:
hrrm alrighty